include/aws/s3/s3_endpoint_resolver.h

```diff
@@ -0,0 +1,27 @@
+#ifndef AWS_S3_ENDPOINT_RESOLVER_H
+#define AWS_S3_ENDPOINT_RESOLVER_H
+
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/s3/private/s3_endpoint_resolver.h>
+#include <aws/s3/s3.h>
+AWS_PUSH_SANE_WARNING_LEVEL
+
+struct aws_endpoints_request_context;
+struct aws_endpoints_rule_engine;
+AWS_EXTERN_C_BEGIN
+
+/**
+ * Creates a new S3 endpoint resolver.
+ * Warning: Before using this header, you have to enable it by
+ * setting AWS_ENABLE_S3_ENDPOINT_RESOLVER=ON
```

Comment:
I would be more specific about this is a CMake config, instead of like env var or something.

Reply:
Thanks, fixed.